dataprep/clean/clean_email.py

```diff
@@ -0,0 +1,514 @@
+    "hotmail.be",
+    "live.be",
+    "skynet.be",
+    "voo.be",
+    "tvcablenet.be",
+    "telenet.be",
+    # rgentinian ISP domains
+    "hotmail.com.ar",
+    "live.com.ar",
+    "yahoo.com.ar",
+    "fibertel.com.ar",
+    "speedy.com.ar",
+    "arnet.com.ar",
+    # Domains used in Mexico
+    "yahoo.com.mx",
+    "live.com.mx",
+    "hotmail.es",
+    "hotmail.com.mx",
+    "prodigy.net.mx",
+    # Domains used in Canada
+    "yahoo.ca",
+    "hotmail.ca",
+    "bell.net",
+    "shaw.ca",
+    "sympatico.ca",
+    "rogers.com",
+    # Domains used in Brazil
+    "yahoo.com.br",
+    "hotmail.com.br",
+    "outlook.com.br",
+    "uol.com.br",
+    "bol.com.br",
+    "terra.com.br",
+    "ig.com.br",
+    "itelefonica.com.br",
+    "r7.com",
+    "zipmail.com.br",
+    "globo.com",
+    "globomail.com",
+    "oi.com.br",
+}
+
+NEARBYKEYS = {
+    "a": ["q", "w", "s", "x", "z"],
+    "b": ["v", "g", "h", "n"],
+    "c": ["x", "d", "f", "v"],
+    "d": ["s", "e", "r", "f", "c", "x"],
+    "e": ["w", "s", "d", "r"],
+    "f": ["d", "r", "t", "g", "v", "c"],
+    "g": ["f", "t", "y", "h", "b", "v"],
+    "h": ["g", "y", "u", "j", "n", "b"],
+    "i": ["u", "j", "k", "o"],
+    "j": ["h", "u", "i", "k", "n", "m"],
+    "k": ["j", "i", "o", "l", "m"],
+    "l": ["k", "o", "p"],
+    "m": ["n", "j", "k", "l"],
+    "n": ["b", "h", "j", "m"],
+    "o": ["i", "k", "l", "p"],
+    "p": ["o", "l"],
+    "q": ["w", "a", "s"],
+    "r": ["e", "d", "f", "t"],
+    "s": ["w", "e", "d", "x", "z", "a"],
+    "t": ["r", "f", "g", "y"],
+    "u": ["y", "h", "j", "i"],
+    "v": ["c", "f", "g", "v", "b"],
+    "w": ["q", "a", "s", "e"],
+    "x": ["z", "s", "d", "c"],
+    "y": ["t", "g", "h", "u"],
+    "z": ["a", "s", "x"],
+    " ": ["c", "v", "b", "n", "m"],
+}
+
+STATS = {"cleaned": 0, "null": 0, "overflow": 0, "bad_format": 0, "unknown": 0}
+
+
+def clean_email(
+    df: Union[pd.DataFrame, dd.DataFrame],
+    column: str,
+    split: bool = False,
+    inplace: bool = False,
+    pre_clean: bool = False,
+    fix_domain: bool = False,
+    errors: str = "coerce",
+) -> pd.DataFrame:
+    """
+    This function cleans emails
+    Parameters
+    ----------
+    df
+        pandas or Dask DataFrame
+    column
+        column name
+    split
+        If True, split a column containing username and domain name
+        into one column for username and one column for domain name
+    inplace
+        If True, delete the given column with dirty data, else, create a new
+        column with cleaned data.
+    pre_clean
+        If True, apply basic text clean(like removing whitespaces) before
+        verifying and clean values.
+    fix_domain
+        If True, fix small typos in domain input
+    errors
+        Specify ways to deal with broken value
+        {'ignore', 'coerce'}, default 'coerce'
+        'ignore': just return the initial input
+        #'raise': raise an exception when there is broken value
+        'coerce': set invalid value to NaN
```

Comment:
I think it would be good if we have a consistent docstring for the `errors` parameter. @yxie66 please take a look at the [clean_lat_long errors docstring](https://github.com/sfu-db/dataprep/blob/53059f94b9608ef6162d66a70f51a87db8ebfd51/dataprep/clean/clean_lat_long.py#L120) and let me know how you think we should make this consistent